lib/y2logsstep.pm

```diff
@@ -57,61 +58,130 @@ sub get_to_console() {
+            send_key 'alt-o';
+            wait_still_screen 1;
+        }
+    }
+    else {
+        while (check_screen('dependency-issue', 5)) {
+            send_key 'alt-1';
             send_key 'spc';
-        };
-        send_key 'alt-o';
+            send_key 'alt-o';
+            wait_still_screen 1;
+        }
     }
+    wait_screen_change { send_key 'alt-a' };
 }
 
-# check for dependency issues, if found, drill down to software selection, take a screenshot, then die
-sub check_and_record_dependency_problems {
+# to break dependency issues
+sub break_dependency {
+    assert_screen 'dependency-issue';
+
+    if (check_var('VIDEOMODE', 'text')) {
+        while (check_screen('dependency-issue-text', 5)) {    # repeat it untill all dependency issues are resolved
+            send_key 'alt-s';                                 # Solution
+            while (check_screen('break-not-highlighted', 5)) {
+                send_key 'down';                              # down to option break dependency
+            }
+            send_key 'ret';                                   # select option break dependency
+            send_key 'alt-o';                                 # OK - Try Again
+            wait_still_screen 1;
+        }
+    }
+    else {
+        while (check_screen('dependency-issue', 5)) {
+            send_key 'alt-2';                                 # 2 is the option to break dependency
+            send_key 'spc';                                   # select it
+            send_key 'alt-o';                                 # OK
+            wait_still_screen 1;
+        }
+    }
+    wait_screen_change { send_key 'alt-a' };
+}
+
+# to deal with dependency issues, either work around it, or break dependency to continue with installation
+sub deal_with_dependency_issues {
     my ($self) = @_;
 
-    return unless check_screen("inst-overview-dep-warning", 1);
+    assert_screen 'manual-intervention';
     record_soft_failure 'dependency warning';
+
     if (check_var('VIDEOMODE', 'text')) {
-        send_key 'alt-c';
+        send_key 'alt-c';    # Change
         assert_screen 'inst-overview-options';
-        send_key 'alt-s';
+        send_key 'alt-s';    # Software
     }
     else {
         send_key_until_needlematch 'packages-section-selected', 'tab';
         send_key 'ret';
     }
-
-    assert_screen 'dependancy-issue';    #make sure the dependancy issue is actually showing
-
     if (get_var("WORKAROUND_DEPS")) {
-        $self->record_dependency_issues;
-        wait_screen_change {
-            send_key 'alt-a';
-        };
+        $self->workaround_dependency_issues;
+    }
+    elsif (get_var 'BREAK_DEPS') {
+        $self->break_dependency;
+    }
+    else {
+        die 'Dependency Problems';
+    }
+    if (check_screen('dependency-issue-fixed', 2)) {
+        send_key 'alt-a';    # Accept
+    }
+    wait_still_screen 2;
+
+  DO_CHECKS:
+    while (check_screen('accept-licence', 2)) {
```

Comment:
I am not convinced that the `check_screen` approach is as stable as the multi-tag assert. Just image a SUT being slightly more busy than usual and it would take more than 2 seconds to show up that dialog. You would fail. I don't agree with this change here. I need better reasons to be convinced that an approach with check_screen is "more reliable" as you state